src/backend/marsha/account/tests/test_views.py

```diff
@@ -0,0 +1,236 @@
+        cls.existing_user = UserFactory(username="existing_username")
+
+    def _login_user(self):
+        """Authenticate the user to be logged in."""
+        self.client.post(
+            reverse("account:login"),
+            {"username": self.existing_user.username, "password": "password"},
+        )
+        self.assertIsAuthenticated()
+
+    def _generate_uid_token_for_password_reset(self):
+        """Generate reset credentials to be used in the reset password link."""
+        return {
+            "uid": urlsafe_base64_encode(force_bytes(self.existing_user.pk)),
+            "token": default_token_generator.make_token(self.existing_user),
+        }
+
+    def assertIsAuthenticated(self):
+        """Assert the user is authenticated."""
+        user = auth_get_user(self.client)
+        self.assertTrue(user.is_authenticated)
+
+    def assertIsNotAuthenticated(self):
+        """Assert the user is *not* authenticated."""
+        user = auth_get_user(self.client)
+        self.assertFalse(user.is_authenticated)
+
+
+class AccountViewTestCase(BaseAuthenticationTestCase):
+    """Test case for the full login/logout process."""
+
+    def test_login_get(self):
+        """Assert the login view is reachable when not logged in."""
+        response = self.client.get(
+            reverse("account:login"),
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Username")
+        self.assertContains(response, "Password")
+        self.assertContains(response, "Forgotten your password or username?")
+
+    def test_login_post_fails(self):
+        """Assert the login view displays an error for non-matching username/password."""
+        response = self.client.post(
+            reverse("account:login"),
+            {"username": self.existing_user.username, "password": "wrong_password"},
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Username")
+        self.assertContains(response, "Password")
+        self.assertContains(response, "Forgotten your password or username?")
+        self.assertContains(
+            response,
+            "Please enter a correct username and password. "
+            "Note that both fields may be case-sensitive.",
+        )
+        self.assertIsNotAuthenticated()
+
+    def test_login_post_success(self):
+        """Assert the login view redirects properly to the main page."""
+        response = self.client.post(
+            reverse("account:login"),
+            {"username": self.existing_user.username, "password": "password"},
+        )
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, "/")
+        self.assertIsAuthenticated()
+
+    def test_logout_post_success(self):
+        """Assert the logout view redirects properly to the main page."""
+        self._login_user()
+
+        response = self.client.post(reverse("account:logout"))
```

Comment:
It is weird to have both `POST` and `GET` don't you think ? What is the reason to have both ?